datagateway_api/src/datagateway_api/database/models.py

```diff
@@ -350,6 +374,210 @@ class DATACOLLECTIONPARAMETER(Base, EntityHelper, metaclass=EntityMeta):
+    modId = Column("MOD_ID", String(255), nullable=False)
+    modTime = Column("MOD_TIME", DateTime, nullable=False)
+    dateType = Column("DATE_TYPE", String(255), nullable=False)
+    date = Column("DATE", String(255), nullable=False)
+    datapublicationId = Column(
+        "DATAPUBLICATION_ID", ForeignKey("DATAPUBLICATION.ID"), nullable=False,
+    )
+
+    DATAPUBLICATION = relationship(
+        "DATAPUBLICATION",
+        primaryjoin="DATAPUBLICATIONDATE.datapublicationId == DATAPUBLICATION.id",
+        backref="dataPublicationDates",
+    )
+
+
+class DATAPUBLICATIONFUNDING(Base, EntityHelper, metaclass=EntityMeta):
+    __tablename__ = "DATAPUBLICATIONFUNDING"
+    __singularfieldname__ = "dataPublicationFunding"
+    __pluralfieldname__ = "dataPublicationFundings"
+    __table_args__ = (
+        Index("UNQ_DATAPUBLICATIONFUNDING_0", "DATAPUBLICATION_ID", "FUNDING_ID"),
+    )
+
+    id = Column("ID", BigInteger, primary_key=True)
+    createId = Column("CREATE_ID", String(255), nullable=False)
+    createTime = Column("CREATE_TIME", DateTime, nullable=False)
+    modId = Column("MOD_ID", String(255), nullable=False)
+    modTime = Column("MOD_TIME", DateTime, nullable=False)
+    datapublicationId = Column(
+        "DATAPUBLICATION_ID", ForeignKey("DATAPUBLICATION.ID"), nullable=False,
+    )
+
+    fundingId = Column("FUNDING_ID", ForeignKey("FUNDINGREFERENCE.ID"), nullable=False)
+
+    DATAPUBLICATION = relationship(
+        "DATAPUBLICATION",
+        primaryjoin="DATAPUBLICATIONFUNDING.datapublicationId == DATAPUBLICATION.id",
+        backref="dataPublicationFundings",
+    )
+
+    FUNDINGREFERENCE = relationship(
+        "FUNDINGREFERENCE",
+        primaryjoin="DATAPUBLICATIONFUNDING.fundingId == FUNDINGREFERENCE.id",
+        backref="dataPublicationFundings",
+    )
+
+
+class DATAPUBLICATIONTYPE(Base, EntityHelper, metaclass=EntityMeta):
+    __tablename__ = "DATAPUBLICATIONTYPE"
+    __singularfieldname__ = "dataPublicationType"
+    __pluralfieldname__ = "dataPublicationTypes"
+    __table_args__ = (Index("UNQ_DATAPUBLICATIONTYPE_0", "FACILITY_ID", "NAME"),)
+
+    id = Column("ID", BigInteger, primary_key=True)
+    createId = Column("CREATE_ID", String(255), nullable=False)
+    createTime = Column("CREATE_TIME", DateTime, nullable=False)
+    modId = Column("MOD_ID", String(255), nullable=False)
+    modTime = Column("MOD_TIME", DateTime, nullable=False)
+    name = Column("NAME", String(255), nullable=False)
+    description = Column("DESCRIPTION", String(255), nullable=False)
+    datapublicationId = Column(
+        "DATAPUBLICATION_ID", ForeignKey("DATAPUBLICATION.ID"), nullable=False,
+    )
+
+    facilityId = Column("FACILITY_ID", ForeignKey("FACILITY.ID"), nullable=False)
+
+    DATAPUBLICATION = relationship(
+        "DATAPUBLICATION",
+        primaryjoin="DATAPUBLICATIONTYPE.datapublicationId == DATAPUBLICATION.id",
+        backref="dataPublicationTypes",
```

Comment:
The correct backref is `type`, please can this be changed?

Reply:
Yes, changed